plugins/parsers/json_v2/parser.go

```diff
@@ -703,9 +703,8 @@ func (p *Parser) convertType(input gjson.Result, desiredType string, name string
 func (p *Parser) checkResult(result gjson.Result, path string, optional bool) (bool, error) {
 	if !result.Exists() {
 		if optional {
-			// If path is marked as optional don't error if path doesn't return a result
 			p.Log.Debugf("the path %q doesn't exist", path)
-			return true, nil
+			return true, fmt.Errorf("the path %q doesn't exist", path)
```

Comment:
Can we please add a comment here saying that the error is required in order for the `skip` to take effect!? Otherwise someone will change this (because honestly speaking this should not be an error) and will break the code...

Reply:
Is it cleaner to re-write this function as:

```
// Check if gjson result exists and if a non-existing gjson result can be skipped over
func (p *Parser) checkResult(result gjson.Result, path string, optional bool) (bool, error) {
	if !result.Exists() {
		p.Log.Debugf("the path %q doesn't exist", path)
		return optional, fmt.Errorf("the path %q doesn't exist", path)
	}
	
	return false, nil
}
```

This makes the relationship between the boolean return value and ``optional`` more explicit.

Reply:
I really love this!